x-pack/plugins/elastic_assistant/server/routes/register_routes.ts

```diff
@@ -61,9 +60,8 @@ export const registerRoutes = (
   findUserConversationsRoute(router);
 
   // Knowledge Base Setup
-  deleteKnowledgeBaseRoute(router);
-  getKnowledgeBaseStatusRoute(router, getElserId);
-  postKnowledgeBaseRoute(router, getElserId);
+  getKnowledgeBaseStatusRoute(router);
+  postKnowledgeBaseRoute(router);
```

Comment:
Since data streams are initiated on plugin start, there's no need to support a delete API any longer, so not registering this internal API anymore. Also, kb routes don't need `getElserId` anymore as the `kbDataClient` encapsulates it.